go.mod

```diff
@@ -38,15 +38,15 @@ require (
 	github.com/vincent-petithory/dataurl v1.0.0
 	golang.org/x/net v0.10.0
 	golang.org/x/time v0.2.0
-	k8s.io/api v0.27.3
-	k8s.io/apiextensions-apiserver v0.27.3
-	k8s.io/apimachinery v0.27.3
-	k8s.io/client-go v0.27.3
-	k8s.io/code-generator v0.27.3
-	k8s.io/component-base v0.27.3
+	k8s.io/api v0.27.4
+	k8s.io/apiextensions-apiserver v0.27.4
+	k8s.io/apimachinery v0.27.4
+	k8s.io/client-go v0.27.4
+	k8s.io/code-generator v0.27.4
+	k8s.io/component-base v0.27.4
 	k8s.io/kubectl v0.27.3
 	k8s.io/kubelet v0.27.3
-	k8s.io/kubernetes v1.27.3
+	k8s.io/kubernetes v1.15.0-alpha.0
```

Comment:
Don't understand why we are switching to an older version here?

Reply:
Hmm I am a little unsure why this is the case as well. The issue was the version of github.com/openshift/library-go was pulling in packages that were dependent on the outdated package in question.

It would seem updating this package downgrades kubernetes
```
$ go get github.com/openshift/library-go
go: upgraded github.com/openshift/api v0.0.0-20230613151523-ba04973d3ed1 => v0.0.0-20230807132801-600991d550ac
go: upgraded github.com/openshift/library-go v0.0.0-20230614142803-865e70cc6b32 => v0.0.0-20230809121909-d7e7beca5bae
go: upgraded k8s.io/api v0.27.3 => v0.27.4
go: upgraded k8s.io/apiextensions-apiserver v0.27.3 => v0.27.4
go: upgraded k8s.io/apimachinery v0.27.3 => v0.27.4
go: upgraded k8s.io/apiserver v0.27.3 => v0.27.4
go: upgraded k8s.io/client-go v0.27.3 => v0.27.4
go: upgraded k8s.io/code-generator v0.27.3 => v0.27.4
go: upgraded k8s.io/component-base v0.27.3 => v0.27.4
go: upgraded k8s.io/kube-aggregator v0.27.1 => v0.27.4
go: downgraded k8s.io/kubernetes v1.27.3 => v1.15.0-alpha.0
go: upgraded sigs.k8s.io/kube-storage-version-migrator v0.0.4 => v0.0.6-0.20230721195810-5c8923c5ff96
```

It doesn't appear that you can directly go get a newer version of kubernetes due to the go.mod in that repo specifying unknown revisions
```
$ go get k8s.io/kubernetes
go: k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0
$ go get k8s.io/kubernetes@v1.27.3
go: k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0

```

So I assume go automatically gets the latest version of kubernetes it can resolve, which is 1.15.0-alpha before the go.mod file was added to k8s.io/kubernetes.

It looks like this is a known issue https://github.com/kubernetes/kubernetes/issues/79384, and actually intentional, as they state using k8s.io/kubernetes module as a library is explicitly not supported https://github.com/kubernetes/kubernetes/#to-start-using-k8s:~:text=k8s.io/kubernetes%20module.

A work around would be to add a replace directive in go.mod for every single library referenced in kubernetes. Or removing the dependency on kuberentes.

Reply:
Hi @sinnykumari updated to remove the downgrade of kubernetes

Reply:
Aahh looks like make unit-test is failing though, will need to investigate which upgrade might have broken things